src/Microsoft.VisualStudio.Threading.Analyzers/VSTHRD201CancelAfterSwitchToMainThreadAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+﻿/********************************************************
+*                                                        *
+*   © Copyright (C) Microsoft. All rights reserved.      *
+*                                                        *
+*********************************************************/
+
+namespace Microsoft.VisualStudio.Threading.Analyzers
+{
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Threading;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Operations;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class VSTHRD201CancelAfterSwitchToMainThreadAnalyzer : DiagnosticAnalyzer
+    {
+        public const string Id = "VSTHRD201";
+
+        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            Id,
+            title: Strings.VSTHRD201_Title,
+            messageFormat: Strings.VSTHRD201_MessageFormat,
```

Comment:
💡 These should use `LocalizableResourceString`

Reply:
Why? How is that better?

Reply:
@mavasani?

Reply:
It allows the diagnostic message to be formatted based on the host's culture where the diagnostics are reported. For example, it is likely that the analyzers execute on a process with a different culture and it's reported diagnostics are serialized to another process/machine with a different culture where the diagnostics are reported.

Reply:
Thanks. In this case we don't even format the message (no formatting args).

Reply:
@AArnott the concern here is the point where the localized resources are evaluated. `LocalizableResourceString` defers the evaluation of these strings to the culture at the time of presentation, which accounts for the possibility of getting serialized and moving to a different process (or possibly even a different machine).

Reply:
Ah, I get it now. I filed an issue to track for all the analyzers.